Packs/SekoiaXDR/Integrations/SekoiaXDR/SekoiaXDR.py

```diff
@@ -442,6 +466,59 @@ def filter_dict_by_keys(input_dict: dict, keys_to_keep: list) -> dict:
     return {key: value for key, value in input_dict.items() if key in keys_to_keep}
 
 
+def fetch_alerts_with_pagination(
+    client: Client,
+    alert_status: str | None,
+    alert_urgency: str | None,
+    alert_type: str | None,
+    max_results: int | None,
+    alerts_created_at: str | None,
+    alerts_updated_at: str | None,
+    sort_by: str | None,
+) -> List[Dict[str, Any]]:
+    """
+    Fetches alerts from the Sekoia XDR API with pagination support.
+    This function retrieves alerts in batches, allowing for efficient handling of large datasets.
+    Args:
+        client (Client): Sekoia XDR client to use.
+        alert_status (str): status of the alert to search for.
+        alert_urgency (str): alert urgency range to search for. Format: "MIN_urgency,MAX_urgency". i.e: 80,100.
+        alert_type (str): type of alerts to search for.
+        max_results (int): Maximum numbers of incidents per fetch.
+        alerts_created_at (str): The date range to search for alerts.
+    Returns:
+        dict: List of alerts retrieved from the API.
+    """
+
+    final_alerts = []
+    offset = 0
+    total_alerts = None
```

Comment:
```suggestion
    total_alerts = 0
```
Let's keep the variable type.

Reply:
Yes agree on that